.github/workflows/all-green.yml

```diff
@@ -0,0 +1,21 @@
+name: Check Pull Request CI Status
+
+on:
+  pull_request:
+    types:
+      - opened
+      - synchronize
+      - reopened
+
+permissions:
+  checks: read
+  statuses: read
+
+jobs:
+  all-jobs-are-green:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Run Ensure CI Success
+        uses: DataDog/ensure-ci-success@a874030af0d650da8864aff94b8b42f83e88a0b5
+        with:
+          initial-delay-seconds: "1000"  # smoke tests takes approx 20 mn to finish
```

Comment:
Assuming that this job is supposed to run after all other jobs have finished (according to the [docs](https://github.com/DataDog/ensure-ci-success)), why not have a higher timeout here? `1000` seconds is about `17` minutes, which is less than the time required for smoke tests to run (and also much less than the runtime of other tests like `Orchestrion` or `System Tests`).

Is there a way to directly call this workflow immediately after all other jobs have finished? The delay might give flaky results and/or not capture all the tests being run (what if we wait a while for a runner, so then tests cannot finish before this job is called?). Or am I misunderstanding how this delay works?

Reply:
Maybe something like this could work better? https://github.com/marketplace/actions/wait-on-check

Reply:
ouch, gitlab/benchmark : 28 minutes ...

Reply:
> And the action from @darccio is also good workaround.

Actually, under the hood, setting a bigger `max-retries` will achieve exatcly the same than `wait-on-check`, without the need to give an arbitrary check name (and will work with gitlab, `wait-on-check` can't  see gitlab, as it uses an API not used by gitlab (check-runs))

Reply:
The benchmarks job definitely takes the longest, but do we want to include that in our Green CI policy? We can see regressions sometimes, which might cause a failing benchmark test, but we may decide to ignore those on a case by case basis (ie introducing a new feature that drastically changes how the tracer works; slower, but necessary). 

Besides that, I did some poking and it seems that the slowest pipelines are Smoke Tests, Orchestrion, and Main Branch. The longest runtime I've seen was ~25 min. 

> Actually, under the hood, setting a bigger max-retries will achieve exatcly the same than a bigger max-retries

@cbeauchesne can you clarify what you mean by this?

Reply:
haha, my brain is melting .... 

"setting a bigger max-retries will achieve exatcly the same ~than a bigger max-retries~ as `wait-on-check`, without the need to give an arbitrary check name (and will work with gitlab"

`wait-on-check` waits for **one** job to be a finished. While `ensure-ci-success` waits for **all** jobs to be finished (and set it's conclusion based on all conclusions.

So basically, set `initial-delay` to 0, and lot of retries, you'll have the same as `wait-on-check`.

The only reason to have `initial-delay` to something is to limit API calls, hitting some github rate limit. But we can definitly give it a try with `0`

Reply:
Makes sense, thank you! Let's keep the `initial-delay` as it is now so we can avoid stealing resources from our runners and pushing up costs. What do you think?

Reply:
Wouldn't be interesting to lower the initial delay so `maxRetries * initialDelaySeconds` sums up more than the expected run time? In this way, improving CI times don't require updating often this workflow.

Also, I'd consider an exponential backoff strategy for retries in `ensure-ci-success`.

Reply:
@hannahkm -> actually, changing initial delay won't impact cost, as it spawns the runner anyway, and sleep. But TBH, it's only one job, will never scale up, so it's negligible compared to dozains of other jobs. This purpose of this delay is to avoid hitting github API rate limiter.

> Wouldn't be interesting to lower the initial delay so maxRetries * initialDelaySeconds sums up more than the expected run time?

@darccio sure, let me do that

> Also, I'd consider an exponential backoff strategy for retries in ensure-ci-success.

If you're ok, I'll keep this one for later :)